public/js/grid/timetable_util.js

```diff
@@ -121,12 +121,16 @@ function restoreFromCookies() {
     var selectedCourseCookie = getCookie('selected-courses');
     var selectedSectionCookie = getCookie('selected-lectures');
 
-    if (selectedCourseCookie.length === 0) {
-        selectedCourseCookie = [];
+    if (selectedCourseCookie === undefined || selectedCourseCookie.length === 0) {
+        selectedCourseCookie = '[]';
+    } else {
+        selectedCourseCookie = '["' + selectedCourseCookie.split('_').join('","') + '"]';
```

Comment:
I've implemented and tested `split('_')`. As far as I can tell, everything is working fine.